roles/fio_distributed/tasks/main.yml

```diff
@@ -100,7 +106,7 @@
       name: "{{ meta.name }}"
       namespace: "{{ operator_namespace }}"
       status:
-        state: StartingClient
+        state: WaitingForCacheDroppers
```

Comment:
There should be 2 when's to update the status. As it stands we will always drop into the waiting for cache droppers block even if workload_args.drop_cache_kernel is not defined.

Reply:
You are correct that it always enters that state, however, if no cache dropping was requested then it immediately leaves that state (see last task in the block).   This keeps the code path simple.   What's the harm in doing it this way?  More complicated to make it branch to cache drop state only if cache dropping was requested.   Wish this entire thing wasn't a state machine.

Reply:
Unless i'm misreading it, it would run all the code from line 121 through 170. which includes creating a configmap, which isn't necessarily a big deal but its creating it for no use.

Reply:
@dry923  good point, I'll try and rethink the extra state.

Reply:
This problem is solved by re-implementing cache dropper as a role outside of any 1 benchmark, and using wait_for_daemonset.py so that no state is needed.